pkgs/by-name/rs/rstudio/package.nix

```
@@ -217,6 +217,14 @@ stdenv.mkDerivation rec {
   ];
 
   postPatch = ''
+    # fix hardcoded paths to /usr/bin/which
+    substituteInPlace \
+    src/node/desktop/src/main/detect-r.ts \
+    src/node/desktop/src/main/gwt-callback.ts \
+    src/cpp/session/modules/clang/CodeCompletion.cpp \
+    src/cpp/core/system/PosixSystemTests.cpp \
+    --replace-fail "/usr/bin/which" "which"
```

```suggestion
    --replace-fail "/usr/bin/which" "${lib.getExe which}"
```

For even better purity?